calculator/calculator.java

```
@@ -0,0 +1,112 @@
+
+        ///Calculater project
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Welcome to Simple Calculator!");
+
+        while (true) {
+            // Ask user for the operation type
+            System.out.println("Enter operation type (+, -, *, /) or 'exit' to quit: ");
+            String operationType  = scanner.next().trim();
+
+            // Exit condition
+            if (operationType.equalsIgnoreCase("exit")) {
+                System.out.println("Exiting the calculator. Goodbye!");
+                break;
+            }
+
+            // Read numbers from the user
+            System.out.println("Enter numbers separated by spaces: ");
+            scanner.nextLine(); // Consume newline left-over
+            String[] numberInputs = scanner.nextLine().split("\\s+");
+
+            if (numberInputs.length < 2) {
+                System.out.println("Please enter at least two numbers.");
+                continue;
+            }
+
+            double[] numbers = new double[numberInputs.length];
+            try {
+                for (int i = 0; i < numberInputs.length; i++) {
+                    numbers[i] = Double.parseDouble(numberInputs[i]);
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid number format. Please enter valid numbers.");
+                continue;
+            }
+
+            double result = 0;
+            boolean validOperation = true;
+
+            switch (operationType) {
+                case "+":
+                    result = add(numbers);
+                    break;
+                case "-":
+                    result = subtract(numbers);
+                    break;
+                case "*":
+                    result = multiply(numbers);
+                    break;
+                case "/":
+                    result = divide(numbers);
+                    break;
+                default:
```

You need to throw an error to the user at the time when the user gave an invalid operation. Instead your app is getting the operands and then throws the error. Because the flow I said will makes the user experience better.